### PR TITLE
Upgrading to samza 1.0

### PR DESCRIPTION
**Pull request by rmatharu**
* Adding WikipediaStatsStreamTaskApplication, WikipediaParserStreamTaskApplication, WikipediaFeedStreamTaskApplication
* Removing task.inputs, and task.class from config. 

* Adding app.class.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * under the License.
  */
 
-SAMZA_VERSION=1.0.0-SNAPSHOT
+SAMZA_VERSION=1.0.0
```
Comment (prateekm) [inline gradle.properties]: Should be 1.0.1-SNAPSHOT on latest.

File: src/main/config/wikipedia-feed.properties [hunk 2]
```diff
@@ -23,22 +23,7 @@ job.name=wikipedia-feed
...
-
-# Kafka System
-systems.kafka.samza.factory=org.apache.samza.system.kafka.KafkaSystemFactory
-systems.kafka.samza.msg.serde=json
-systems.kafka.consumer.zookeeper.connect=localhost:2181/
-systems.kafka.producer.bootstrap.servers=localhost:9092
+app.class=samza.examples.wikipedia.task.application.WikipediaFeedStreamTaskApplication
 
 # Job Coordinator
 job.coordinator.system=kafka
```
Comment (prateekm) [inline src/main/config/wikipedia-feed.properties]: Can you remove this (everywhere) and set the default system descriptor in the TaskApplication instead?

Comment (prateekm) [reply src/main/config/wikipedia-feed.properties]: Can you also update teh comment below if necessary. Seems outdated.

File: src/main/config/wikipedia-parser.properties [hunk 3]
```diff
@@ -23,16 +23,13 @@ job.name=wikipedia-parser
...
-task.class=samza.examples.wikipedia.task.WikipediaParserStreamTask
-task.inputs=kafka.wikipedia-raw
+app.class=samza.examples.wikipedia.task.application.WikipediaParserStreamTaskApplication
 
 # Serializers
-serializers.registry.json.class=org.apache.samza.serializers.JsonSerdeFactory
 serializers.registry.metrics.class=org.apache.samza.serializers.MetricsSnapshotSerdeFactory
 
 # Systems
 systems.kafka.samza.factory=org.apache.samza.system.kafka.KafkaSystemFactory
```
Comment (prateekm) [inline src/main/config/wikipedia-parser.properties]: Are these kafka system configs still required?

File: src/main/config/wikipedia-stats.properties [hunk 4]
```diff
@@ -40,7 +39,6 @@ serializers.registry.integer.class=org.apache.samza.serializers.IntegerSerdeFact
 
 # Systems
 systems.kafka.samza.factory=org.apache.samza.system.kafka.KafkaSystemFactory
```
Comment (prateekm) [inline src/main/config/wikipedia-stats.properties]: Is the json serde above used anywhere?

File: src/main/config/wikipedia-stats.properties [hunk 5]
```diff
@@ -40,7 +39,6 @@ serializers.registry.integer.class=org.apache.samza.serializers.IntegerSerdeFact
...
-systems.kafka.samza.msg.serde=json
 systems.kafka.consumer.zookeeper.connect=localhost:2181/
 systems.kafka.consumer.auto.offset.reset=largest
 systems.kafka.producer.bootstrap.servers=localhost:9092
```
Comment (prateekm) [inline src/main/config/wikipedia-stats.properties]: Would be good to change the stores to tables in this example and use table descriptors. Can do separately if it's too much work.

File: src/main/java/samza/examples/wikipedia/task/application/WikipediaFeedStreamTaskApplication.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import org.apache.samza.application.descriptors.TaskApplicationDescriptor;
+import org.apache.samza.serializers.JsonSerde;
+import org.apache.samza.system.kafka.descriptors.KafkaSystemDescriptor;
+import org.apache.samza.task.StreamTaskFactory;
+import samza.examples.wikipedia.system.descriptors.WikipediaInputDescriptor;
+import samza.examples.wikipedia.system.descriptors.WikipediaSystemDescriptor;
+import samza.examples.wikipedia.task.WikipediaStatsStreamTask;
+
+
+public class WikipediaFeedStreamTaskApplication implements TaskApplication {
```
Comment (prateekm) [inline src/main/java/samza/examples/wikipedia/task/application/WikipediaFeedStreamTaskApplication.java]: WikipediaFeedTaskApplication

File: src/main/java/samza/examples/wikipedia/task/application/WikipediaFeedStreamTaskApplication.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+            .withProducerBootstrapServers(KAFKA_PRODUCER_BOOTSTRAP_SERVERS)
+            .withDefaultStreamConfigs(KAFKA_DEFAULT_STREAM_CONFIGS);
+
+    // Set the inputs
+    taskApplicationDescriptor.withInputStream(wikipediaInputDescriptor);
+    taskApplicationDescriptor.withInputStream(wiktionaryInputDescriptor);
+    taskApplicationDescriptor.withInputStream(wikiNewsInputDescriptor);
+
+    taskApplicationDescriptor.withOutputStream(
+        kafkaSystemDescriptor.getOutputDescriptor("wikipedia-stats", new JsonSerde<>()));
```
Comment (prateekm) [inline src/main/java/samza/examples/wikipedia/task/application/WikipediaFeedStreamTaskApplication.java]: Prefer extracting the OutputDescriptor to a field, next to the input descriptors.

File: src/main/java/samza/examples/wikipedia/task/application/WikipediaParserStreamTaskApplication.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+    // Define a system descriptor for Kafka
+    KafkaSystemDescriptor kafkaSystemDescriptor = new KafkaSystemDescriptor("kafka")
+        .withConsumerZkConnect(KAFKA_CONSUMER_ZK_CONNECT)
+        .withProducerBootstrapServers(KAFKA_PRODUCER_BOOTSTRAP_SERVERS)
+        .withDefaultStreamConfigs(KAFKA_DEFAULT_STREAM_CONFIGS);
+
+
+    // Input descriptor for the wikipedia-edits topic
+    KafkaInputDescriptor kafkaInputDescriptor =
```
Comment (prateekm) [inline src/main/java/samza/examples/wikipedia/task/application/WikipediaParserStreamTaskApplication.java]: Should add output descriptor too.

File: src/main/java/samza/examples/wikipedia/task/application/WikipediaStatsStreamTaskApplication.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  public void describe(TaskApplicationDescriptor taskApplicationDescriptor) {
+
+    // Define a system descriptor for Kafka
+    KafkaSystemDescriptor kafkaSystemDescriptor = new KafkaSystemDescriptor("kafka")
+        .withConsumerZkConnect(KAFKA_CONSUMER_ZK_CONNECT)
+        .withProducerBootstrapServers(KAFKA_PRODUCER_BOOTSTRAP_SERVERS)
+        .withDefaultStreamConfigs(KAFKA_DEFAULT_STREAM_CONFIGS);
+
+    // Input descriptor for the wikipedia-edits topic
+    KafkaInputDescriptor kafkaInputDescriptor =
```
Comment (prateekm) [inline src/main/java/samza/examples/wikipedia/task/application/WikipediaStatsStreamTaskApplication.java]: Should add output descriptor too.

**Comment by rmatharu** [comment]
Merged

**Pull request closed without merging**